### ISSUE TITLE
How to address QAQC for remote sensing data

### ISSUE THREAD

**Issue by mcgregorian1** [EcoClimLab/growth_phenology#109]
@camerondow35 

I realized when going through the EVI2 / phenometrics Earth Engine code that the application of the QAQC is easiest if done outside of GEE (in the sense that, if I did need to do it in GEE I would need to fully overhaul the script).

I created an R-script to go through and remove values based on QAQC for the phenology. Really, all this results in is one Greenup value needing to be removed because of poor quality, but the script is robust for if other sites needed to be checked. I'll add the files here to this issue for you to review.
- Note that values were kept if they were "good" or "best" quality, and removed otherwise.

Two questions for you:
- Where would you like the script file to be added? 
- Did you use the EVI2 values directly from the table output by Earth engine? Those also have QA flags, but the descriptions in GEE aren't adequate enough by themselves for me to filter.

Note this file is not with the updated coordinates yet

[scbiPhenFiltered.csv](https://github.com/EcoClimLab/growth_phenology/files/8010757/scbiPhenFiltered.csv)

**Comment by camerondow35** [comment]
Great, thanks @mcgregorian1! 

> Where would you like the script file to be added?

The script can go in the folder with the [other scripts](https://github.com/EcoClimLab/growth_phenology/tree/master/RScripts)

> Did you use the EVI2 values directly from the table output by Earth engine? Those also have QA flags, but the descriptions in GEE aren't adequate enough by themselves for me to filter.

I believe I used the EVI2 values from the csv's you provided [here](https://github.com/EcoClimLab/growth_phenology/blob/master/Data/Leaf%20phenology/scbiEVI2.csv), which I assume are directly from earth engine.

**Comment by mcgregorian1** [comment]
Thanks! Here are some updates.

MCD12Q2 (phenometric dates)
- As mentioned in my previous comment, the QA flags here are relatively simple, and we ended up removing 1 Greenup value due to poor quality. Otherwise, we are keeping observations that are either "good" or "best".

MCD43A4 (EVI2)
- The GEE script calculates EVI2 from the band values given by MCD43A4. The QA flags here are simplified to be either 0 (good) or 1 (other). If an observation has a QA flag of 1, then the description on GEE says to check the full QA, which is in the product of MCD43A2. Looking at that, it turns out that the simplified QA of A4 matches pretty well with A2, in that anything with a 1 in A4 is neither "good" nor "best" in A2. 
- All that to say, if we follow the same protocol we use for MCD12Q2 and only use "good" and "best" data, then the QA flags from A4 are satisfactory as is. I guess, I should retract my earlier comment asking why the reviewer brought this up when they were indeed correct.
- **Note** the filtering process yields *much* less data than you were using before, so make sure to re-make the plots you were using.

Scripts
- In any case, I have made 2 scripts dealing with this. The script `scbiPhenQA.R` is the main one, taking arguments for the file path of the GEE-output-data and a file path for where to save the filtered csv. As of now, I've saved them as separate files in case something comes up so that way I don't have to wait for Earth Engine to output the csv again.
- The second script `scbiPhenQA_functions.R` has the associated functions for the first script, and they are sourced in the first script. If you think these would be better combined, that's fine.
- I have included the notes above in the R files, so hopefully anyone who is looking at this will be able to fully understand this without further explanation. There I have included the link to the GEE script as well as the link to the ForestGEOsites csv as a GEE asset (necessary for running the script as is).

General
- I've uploaded the files and have put them in the [scripts folder]() as well as the [growth_phenology]() folder.
